### PR TITLE
[SYCL][Matrix][E2E] make joint_matrix_transposeC_impl.hpp simpler: just joint_matrix_load&joint_matrix_store for colC

### PR DESCRIPTION
**Pull request by yubingex007-a11y**

### CODE REVIEW

**Comment by yubingex007-a11y** [comment]
I think we can merge this PR after my backend patch is merged

**Review by dkhaldi — APPROVED** [review]
LGTM

**Review by YuriPlyakhin — CHANGES_REQUESTED** [review]

**Comment by YuriPlyakhin** [comment]
Also should XFAIL be updated here as well? SG32/joint_matrix_transposeC.cpp

**Comment by yubingex007-a11y** [comment]
ping @YuriPlyakhin

File: sycl/test-e2e/Matrix/joint_matrix_transposeC_impl.hpp [hunk 1]
```diff
@@ -18,12 +18,9 @@ void matrix_multiply(T1 *C, T2 *A, T2 *B, queue q, unsigned int vnniFactor) {
   size_t NDRangeM = M / TM;
   size_t NDRangeN = N / TN;
 
-  auto pA = address_space_cast<sycl::access::address_space::global_space,
-                               sycl::access::decorated::no>(A);
-  auto pB = address_space_cast<sycl::access::address_space::global_space,
-                               sycl::access::decorated::no>(B);
-  auto pC = address_space_cast<sycl::access::address_space::global_space,
-                               sycl::access::decorated::no>(C);
+  auto pA = multi_ptr<T2, sycl::access::address_space::global_space>(A);
```
[the remark this answers is not in the document]
Comment (yubingex007-a11y) [reply sycl/test-e2e/Matrix/joint_matrix_transposeC_impl.hpp]: done

File: sycl/test-e2e/Matrix/joint_matrix_transposeC_impl.hpp [hunk 2]
```diff
@@ -18,12 +18,9 @@ void matrix_multiply(T1 *C, T2 *A, T2 *B, queue q, unsigned int vnniFactor) {
...
+  auto pB = multi_ptr<T2, sycl::access::address_space::global_space>(B);
```
[the remark this answers is not in the document]
Comment (yubingex007-a11y) [reply sycl/test-e2e/Matrix/joint_matrix_transposeC_impl.hpp]: done

File: sycl/test-e2e/Matrix/joint_matrix_transposeC_impl.hpp [hunk 3]
```diff
@@ -88,8 +75,6 @@ int main() {
   matrix_multiply<float, bfloat16, MATRIX_M, MATRIX_K, MATRIX_K / vnniFactor,
```
[the remark this answers is not in the document]
Comment (yubingex007-a11y) [reply sycl/test-e2e/Matrix/joint_matrix_transposeC_impl.hpp]: done

**Comment by yubingex007-a11y** [comment]
> Also should XFAIL be updated in SG32/joint_matrix_transposeC.cpp as well?

done

**Review by YuriPlyakhin — APPROVED** [review]
LGTM

**Comment by YuriPlyakhin** [comment]
@intel/llvm-gatekeepers , all green, please, merge

**Pull request merged**